### PR TITLE
allow email recipients to be specified with a select2

### PR DESCRIPTION
**Pull request by mjriley**
## Technical Summary
Associated Ticket: https://dimagi-dev.atlassian.net/browse/SAAS-12758
I'm not sure how this worked in the past. In other parts of HQ, multiple inputs like this are done through a Select2 control. It's possible that Select2 was updated from using a text area to using a select control, but I'm not sure. Even if that were the case, we were not importing the required javascript library to turn this field into a select2 control.

Prior to this PR, we had explicit HTML to represent the form, and then verified the inputs via a django form. This PR makes it so that the form itself is the single place in charge of both rendering and verifying. This also provides error feedback to the user.

### Safety story
Locally tested, and there is a minimal test suite. I'd like this to run through QA, just because I don't really understand why the code was the way it was prior. Even though I couldn't find any other references to this code, I want to make sure I'm not breaking any existing email flows, and I want to make sure that the `attach_excel` field wasn't used.

### Automated test coverage

A minimal test suite was created at reports/tests/test_forms to verify that error output given to the user.

### QA Plan

Plan is to test any existing email flows.
[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-5491)

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/standard/__init__.py [hunk 1]
```diff
@@ -40,7 +41,7 @@ def default_report_url(self):
     @property
     def template_context(self):
         context = super().template_context
-        context.update({'user_types': HQUserType.human_readable})
+        context.update({'user_types': HQUserType.human_readable, 'email_form': EmailReportForm()})
```
Comment (biyeun) [inline corehq/apps/reports/standard/__init__.py:44]:
```suggestion
        context.update({
            'user_types': HQUserType.human_readable,
            'email_form': EmailReportForm()
        })
```

File: corehq/apps/reports/static/reports/js/hq_report.js [hunk 2]
```diff
@@ -214,13 +216,13 @@ hqDefine("reports/js/hq_report", [
 
             self.send_to_owner = ko.observable(true);
             self.subject = ko.observable(hqReport.emailDefaultSubject);
-            self.recipient_emails = ko.observable();
+            self.recipient_emails = ko.observableArray();
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/hq_report.js:219]: i have a feeling this not being an `observableArray` might have been the crux of the issue

File: corehq/apps/reports/static/reports/js/standard_hq_report.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ hqDefine("reports/js/standard_hq_report", [
                     // Standard reports
                     var reportOptions = _.extend({}, initialPageData.get('js_options'), {
                         emailSuccessMessage: gettext('Report successfully emailed'),
-                        emailErrorMessage: gettext('An error occurred emailing you report. Please try again.'),
+                        emailErrorMessage: gettext('An error occurred emailing your report. Please try again.'),
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/js/standard_hq_report.js:44]: thank you

File: corehq/apps/reports/views.py [hunk 4]
```diff
@@ -541,21 +541,33 @@ def user(self):
         return self.request.couch_user
 
 
+def querydict_to_dict(query_dict):
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/reports/views.py:544]: I should do some searching to figure out whether our code has to deal with similar situations. I'm leaning towards making it private, as my assumption (based on how this view used to work) is that most of our frontend code submits multiple duplicate parameters as `someParam=one&someParam=two`, compared to (I believe) the standard of `someParam[]=one&someParam[]=two`. I'm not sure if its more confusing to have the standard approach just residing here, or to put the standard, but hardly used approach, in a common utility class

**Comment by minhaminha** [comment]
Looks good! Just curious, but did you test on staging? I noticed the branch wasn't in `staging.yml`.

**Comment by mjriley** [comment]
> Looks good! Just curious, but did you test on staging? I noticed the branch wasn't in `staging.yml`.

Not yet, I was waiting for confirmation that the approach was sound before putting on staging

**Review by biyeun — APPROVED** [review]
great! 🎉

**Review by biyeun — APPROVED** [review]

**Pull request merged**